File: Projects/DeliveryApp/DeliveryApp.Data/Objects/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace DeliveryApp.Data.Objects
{
    public class Order
    {
        [Key]
        public int ID { get; set; }
        //public int CustomerID { get; set; } //Gives id for each customer
        public User Customer { get; set; }  // This is a navigation property, allows us to acces the Customer Property
        //public int PreparerID { get; set; } 
        public User Preparer { get; set; }  // This is a navigation property, allows us to acces the Customer Property
        //public int DriverID { get; set; } 
        public User Driver { get; set; }    // This is a navigation property, allows us to acces the Customer Property
        //public int LocationID { get; set; } 
        public Restaurant Location { get; set; }
        public List<OrderItem> OrderItems { get; set; }
        public Order() { }

    }
}

File: Projects/DeliveryApp/ClientApp/src/app/models/order.ts
import { OrderItem } from './orderitem';
import { Restaurant } from './restaurant';

export class Order {
    id: number;
    customerID: number;
    preparerID: number;
    driverID: number;
    locationID: number;
    location: Restaurant;
    orderItems: OrderItem[];

    constructor(customerID: number, preparerID: number, driverID: number, locationID: number) {
        this.customerID = customerID;
        this.preparerID = preparerID;
        this.driverID = driverID;
        this.locationID = locationID;
        this.orderItems = [];
    }
}
